### ISSUE TITLE
fix: Too many requests to `GET /v1/servers/{id}`

### ISSUE THREAD

**Issue by apricote** [hetznercloud/hcloud-cloud-controller-manager#661]
### TL;DR

In some situations hcloud-cloud-controller-manager starts to spam `GET /v1/servers/{id}` for a subset of the nodes in the cluster every few seconds.

### Expected behavior

I would expect hccm to only send a single request per `--node-refresh-interval` (defaults to 5 minutes).

### Observed behavior

We get reports from customers that exhaust the rate limit and we can see that hcloud-cloud-controller-manager sends > 1 req/s for `GET /v1/servers/{id}`. The default rate limit is 1 request per second.

A restart of the pod fixes the behaviour.

### Minimal working example

We are not sure how to reproduce this yet.

### Additional information

Looking at some request logs, this seems to affect even very old versions of HCCM (>2 years).

**Comment by lukasmetzner** [comment]
@apricote during the server cache implementation we gathered some additional insights in the node and node lifecycle controller, which lead to a hint that this cache might fix our problem here. Do you think its worth closing this issue and reopening it, if we see similar behavior in customer traffic?

**Comment by apricote** [comment]
Let's keep it open until we have enough upgraded clusters so we can check the metrics and verify that it is no longer happening
